packages/pnpm/src/main.ts

```diff
@@ -205,6 +205,16 @@ export default async function run (argv: string[]) {
     setTimeout(() => {
       // `pnpm install ""` is going to be just `pnpm install`
       const cliArgs = cliConf.argv.remain.slice(1).filter(Boolean)
+
+      // checking for args that should be passed to the command that is specfied in the build script.
+      const argIndex = cliArgs.findIndex((value) => {
```

Comment:
This is how argIndex should be found: 

```ts
const argIndex = cliConf.argv.cooked.indexOf('--') - 1
```

Also, I would give the variable some more meaningful name. `dashDashIndex` or `doubleDashIndex` or `paramsSeparatorIndex`

Reply:
Seems like the right solution is to pass the raw argv to npm when the commands are passed through to npm